src/components/organisms/AuthenticationModal/LoginForm/LoginForm.tsx

```diff
@@ -56,31 +61,43 @@ const LoginForm: React.FunctionComponent<PropsType> = ({
   const onSubmit = async (data: LoginFormData) => {
     if (!venue) return;
     try {
-      if (venue.requiresTicketCode) await axios.get(codeCheckUrl(data.code));
-      if (venue.requiresEmailVerification)
-        await axios.get(codeCheckUrl(data.email));
-
-      const auth = await signIn(data);
-
-      if (
-        auth.user &&
-        (venue.requiresTicketCode || venue.requiresEmailVerification)
-      ) {
-        firebase
-          .firestore()
-          .doc(`userprivate/${auth.user.uid}`)
-          .get()
-          .then((doc) => {
-            if (auth.user && doc.exists) {
-              firebase
-                .firestore()
-                .doc(`userprivate/${auth.user.uid}`)
-                .update({
-                  codes_used: [...(doc.data()?.codes_used || []), data.code],
-                  date_of_birth: data.date_of_birth,
-                });
-            }
-          });
+      await signIn(data);
+
+      if (venue.access?.includes(VenueAccessMode.Codes)) {
+        const result = await checkAccess({
+          venueId: venue.id,
+          code: data.code,
+        });
+        if (result.data === false) {
+          throw new Error("access denied");
+        }
+        console.log("result.data", result.data);
+        console.log(
+          "saving token to",
+          accessTokenKey(venue.id),
+          ": ",
+          result.data.token
+        );
+        setLocalStorageToken(venue.id, result.data.token);
+      }
+      if (venue.access?.includes(VenueAccessMode.Emails)) {
```

Comment:
As above RE: `venue.access` seemingly needing to be an array (yet we changed the type of it I think)